packages/paste-website/src/components/Heading.tsx

```diff
@@ -20,15 +23,22 @@ const AnchoredHeading: React.FC<HeadingProps> = (props) => {
 
     return (
       <Heading {...props} id={id}>
-        {props.children}
-        <StyledAnchorHyperlink href={`#${id}`} data-cy={`anchored-heading-${props.as}`}>
-          #
-        </StyledAnchorHyperlink>
+        <Box as="span" display="flex" alignItems="center">
+          {props.children}
+          <StyledAnchorHyperlink data-cy={`anchored-heading-${props.as}`} href={`#${id}`}>
+            <LinkIcon
+              color="colorTextIcon"
+              decorative={false}
+              title={`${props.children} page anchor`}
```

Comment:
Wasn't sure what would be the best title here, but this seemed the most usable/logical.